app/src/main/res/layout/fragment_add_edit_book.xml

```diff
@@ -0,0 +1,167 @@
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@+id/authorEditText" />
+
+            <Button
+                android:id="@+id/scanIsbnButton"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginEnd="32dp"
+                android:text="Scan"
+                app:layout_constraintBaseline_toBaselineOf="@+id/isbnEditText"
+                app:layout_constraintEnd_toEndOf="parent" />
+
+            <EditText
+                android:id="@+id/isbnEditText"
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="16dp"
+                android:layout_marginTop="8dp"
+                android:layout_marginEnd="16dp"
+                android:digits="0123456789"
+                android:ems="10"
+                android:hint="@string/bookisbn_prompt"
+                android:inputType="number"
+                app:layout_constraintEnd_toStartOf="@+id/scanIsbnButton"
+                app:layout_constraintHorizontal_bias="0.0"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@+id/isbnPrompt" />
+
+            <Button
+                android:id="@+id/populateFromIsbnBtn"
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="16dp"
+                android:layout_marginTop="8dp"
+                android:layout_marginEnd="16dp"
+                android:text="@string/populateFromIsbnBtnText"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@+id/isbnEditText" />
+
+            <TextView
+                android:id="@+id/descriptionPrompt"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="16dp"
+                android:layout_marginTop="16dp"
+                android:text="@string/bookdesc_prompt"
+                app:layout_constraintBottom_toTopOf="@+id/editText4"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@+id/populateFromIsbnBtn" />
+
+            <EditText
+                android:id="@+id/editText4"
+                android:layout_width="0dp"
+                android:layout_height="150dp"
+                android:layout_marginStart="16dp"
+                android:layout_marginEnd="16dp"
+                android:layout_marginBottom="16dp"
+                android:ems="10"
+                android:hint="@string/bookdesc_prompt"
+                android:inputType="textMultiLine"
+                app:layout_constraintBottom_toBottomOf="parent"
```

Comment:
There is no constraint to the top to the descriptionPrompt so this actually looks really bad on my screen.

This is something I would really liked to see fixed if possible since it's annoying.

This is what it looks like in the layout editor:
![alex_constraint](https://user-images.githubusercontent.com/15524738/53677575-785e6780-3c6f-11e9-8887-ed6ebc65d831.PNG)

This is what it looks like on my screen. Note that my screen has a red tint due to an app I installed to reduce eye strain, it's nothing to do with our application. Notice the huge gap between the prompt and the edit text. I think you should really change this.
![screenshot_20190301-215715](https://user-images.githubusercontent.com/15524738/53677584-8ca26480-3c6f-11e9-93fc-14c04e8ec5f8.png)

Reply:
Fixed! Thanks :) 

For reference, it was actually not the constraint issue, but the defaults of edit text setting the gravity to be center left rather than top left. If this happens to anyone else, you can fix it by changing the android:textalign attribute in the edit text to android:gravity="top|left"
Note that you can't just append that since it will complain if you have both textalign and gravity.